src/components/pages/SingleArticle.jsx

```diff
@@ -1,32 +1,75 @@
 import React, { useEffect, useState } from 'react'
 import { useParams } from 'react-router-dom'
 import { Comments } from '../Comments'
-import { fetchSingleArticle, formatDate } from '../../../utils/utils'
+import { fetchSingleArticle, formatDate, patchArticle } from '../../../utils/utils'
 
 export const SingleArticle = () => {
     const { article_id } = useParams()
     const [article, setArticle] = useState('')
     const [isLoading, setIsLoading] = useState(true)
+    const [vote, setVote] = useState(0)
+    const [voteOffset, setVoteOffset] = useState(0)
+
+    const voteCount = Number(vote) + Number(article.votes)
 
     useEffect(() => {
-        fetchSingleArticle(article_id)
-        .then((response) => {
+        fetchSingleArticle(article_id).then((response) => {
             setArticle(response)
             setIsLoading(false)
         })
     }, [])
 
+    useEffect(() => {
+        patchArticle(article_id, {inc_votes: Number(voteOffset)})
+        .catch((error)=>console.log(error, "vote error!"))
```

Comment:
This .catch is where the logic to reverse an optimistic vote should be, and any other error handling logic related to the patchArticle request